psqlextra/partitioning/plan.py

```diff
@@ -36,8 +39,10 @@ def apply(self, using: Optional[str]) -> None:
 
         connection = connections[using or "default"]
 
-        with transaction.atomic():
-            with connection.schema_editor() as schema_editor:
+        with self._migration_context_manager():
+            with connection.schema_editor(
+                atomic=self.config.atomic
+            ) as schema_editor:
```

Comment:
I was thinking that it would be handy to merge these two context managers in one.
Like some `schema_editor` taking an `atomic` parameter to decide if it wraps everything with `transaction.atomic()` or not. Ah it takes already that :trollface: 

Ok let's keep this simple, hopefully it'll be easier to merge upstream